modules/mbedtls/Kconfig.tls-generic

```diff
@@ -567,6 +568,38 @@ config MBEDTLS_PSA_P256M_DRIVER_RAW
 	  Warning: Usage of this Kconfig option is prohibited in Zephyr's codebase.
 	           Users can enable it in case of very memory-constrained devices, but be aware that the p256-m interface is absolutely not guaranted to remain stable over time.
 
+config MBEDTLS_PSA_STATIC_KEY_SLOTS
+	bool "Use statically allocated key buffers to store key material"
+	default y if !MBEDTLS_ENABLE_HEAP
+	help
+	  By default Mbed TLS's PSA Crypto core uses heap memory to store the
+	  key's material for each key slot. This might impose an undesired
+	  requirement to support heap memory and its management code affecting
+	  RAM and ROM footprints at the same time.
+	  Enabling this symbol cause Mbed TLS to pre-allocate all the key slot
+	  buffers that are used to store the key's material at built time, thus
+	  removing the need for heap memory. Each buffer will be sized to
+	  contain the largest asymmetric/symmetric key type enabled in the build
+	  through PSA_WANT symbols.
```

Comment:
This would be true only if you build Mbed TLS with MBEDTLS_PSA_CRYPTO_C enabled but not MBEDTLS_PSA_CRYPTO_CONFIG so that PSA_WANTs are derived/guessed from legacy symbols, but this is not something we do in Zephyr

Reply:
So with how we do it in Zephyr, PSA_WANTs are not enabled by Mbed TLS symbols? It's only the other way around?

Reply:
Since PSA crypto is the defacto crypto API in zephyr, the PSA_WANT_ALG_XXX etc. is the appropriate way forward.

Legacy CONFIGS converted to PSA crypto has code-size and ram size side-effects, and it also makes it difficult to optimize